source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1198,6 +1198,31 @@ TEST(NiFakeServiceTests, NiFakeService_GetAttributeViString_CallsGetAttributeViS
   EXPECT_THAT(response.attribute_value(), ElementsAreArray(attribute_char_array, expected_size));
 }
 
+TEST(NiFakeServiceTests, NiFakeService_AcceptViUInt32Array_CallsAcceptViUInt32Array)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  const unsigned long uInt32Array[] = {0, 1, 4294967293, 4294967294, 4294967295};
+  std::int32_t array_len = 5;
+  EXPECT_CALL(library, AcceptViUInt32Array(kTestViSession, array_len, _))
+      .With(Args<2, 1>(ElementsAreArray(uInt32Array)))
+      .WillOnce(Return(kDriverSuccess));
+
+  ::grpc::ServerContext context;
+  nifake_grpc::AcceptViUInt32ArrayRequest request;
+  request.mutable_vi()->set_id(session_id);
+  for (unsigned int elem : uInt32Array) {
```

Comment:
Use the the iterator-based function that takes `uint32_array.begin()`, and `uin32_array.end()` instead of writing a for loop.

Reply:
using `CopyFrom` function